File: backend/Core/Models/Dtos/Database/FlowAreaPreviewDto.cs
namespace Core.Models.Dtos
{
    /// <summary>
    /// What an area resolves to on this machine right now, with an optional screenshot of it.
    /// Powers the Preview button and, later, the import binding report.
    /// </summary>
    public class FlowAreaPreviewDto
    {
        public bool IsResolved { get; set; }
        public string? ErrorMessage { get; set; }

        public int LocationX { get; set; }
        public int LocationY { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }

        /// <summary>
        /// The DPI of the monitor holding most of it. What a capture inside it records, so the
        /// pixels can follow the monitor on another screen.
        /// </summary>
        public int Dpi { get; set; }

        /// <summary>JPEG bytes, arrives in the renderer as base64.</summary>
        public byte[]? Screenshot { get; set; }
    }
}

File: frontend/src/shared/models/database/flow-area-preview-dto.ts
export interface FlowAreaPreviewDto {
  isResolved: boolean;
  errorMessage?: string;

  locationX: number;
  locationY: number;
  width: number;
  height: number;

  // The DPI of the monitor holding most of it: what a capture inside it records.
  dpi: number;

  // JPEG bytes, arrives as base64.
  screenshot?: string;
}
